spec/models/mixins/purging_mixin_spec.rb

```diff
@@ -1,84 +1,81 @@
+      expect(example_class.pluck(:id)).to match_array(all_ids)
     end
 
-    it "with a date out of range from configuration" do
+    it "with a date out of range" do
+      events # create events
       allow(example_class).to receive(:purge_date).and_return(6.months.ago)
-
-      expect(example_class).to receive(:delete_all).never
-      expect(example_associated_class).to receive(:delete_all).never
-      example_class.purge
+      expect(example_class.purge).to eq(0)
+      expect(example_class.pluck(:id)).to match_array(all_ids)
     end
 
     it "with a date within range" do
-      expect(example_class).to receive(:delete_all).once.and_call_original
-      expect(example_associated_class).to receive(:delete_all).once.and_call_original
-      example_class.purge(purge_date + 1.second)
-      expect(example_class.pluck(:id)).to match_array @unpurged_ids
+      events # create events
+      expect(example_class.purge(purge_date + 1.second)).to eq(3)
+      expect(example_class.pluck(:id)).to match_array unpurged_ids
     end
 
     it "with a date within range from configuration" do
+      events # create events
       allow(example_class).to receive(:purge_date).and_return(purge_date + 1.second)
-
-      expect(example_class).to receive(:delete_all).once.and_call_original
-      expect(example_associated_class).to receive(:delete_all).once.and_call_original
-      example_class.purge
-      expect(example_class.pluck(:id)).to match_array @unpurged_ids
+      expect(example_class.purge).to eq(3)
+      expect(example_class.pluck(:id)).to match_array unpurged_ids
     end
 
     it "with a date covering the whole range" do
-      expect(example_class).to receive(:delete_all).once.and_call_original
-      expect(example_associated_class).to receive(:delete_all).once.and_call_original
-      example_class.purge(Time.now)
+      events # create events
+      expect(example_class.purge(Time.now)).to eq(5)
       expect(example_class.pluck(:id)).to match_array []
     end
 
     it "with a date covering the whole range from configuration" do
+      events # create events
       allow(example_class).to receive(:purge_date).and_return(Time.now)
-
-      expect(example_class).to receive(:delete_all).once.and_call_original
-      expect(example_associated_class).to receive(:delete_all).once.and_call_original
-      example_class.purge
+      expect(example_class.purge(Time.now)).to eq(5)
       expect(example_class.pluck(:id)).to match_array []
     end
 
     it "with a date and a window" do
-      expect(example_class).to receive(:delete_all).twice.and_call_original
-      expect(example_associated_class).to receive(:delete_all).twice.and_call_original
-      example_class.purge(purge_date + 1.second, 2)
-      expect(example_class.pluck(:id)).to match_array @unpurged_ids
+      events # create events
+      expect(example_class.purge(purge_date + 1.second, 2)).to eq(3)
+      expect(example_class.pluck(:id)).to match_array all_ids[-2, 2]
```

Comment:
Minor, but a simpler and more readable way to do `all_ids[-2, 2]` is `all_ids.last(2)`